### ISSUE TITLE
Feature Request: support ansible style inventory|group|host variables and templating

### ISSUE THREAD

**Issue by stephencheng** [go-task/task#208]
One of common problem to use make/ant/ansible/puppet to build is to deal with different set of variables based on its context. I found ansible handles it well to categorise them into all | group_var | host_var, so that the vars are context aware.

This is good for Task to be used as a provisioning tool. If an easy templating mechanism is builtin that's even better.

I am wishing to see a superset of tool to have some type of integration of: 
- task+[group var|inventory|templating(go template|jinja2)]
- mage [to handle golang talking to various apis]
- https://github.com/pressly/sup [local/remote provision]
- https://github.com/direnv/direnv [deal with loading env vars]

So that ansible could be replaced

**Comment by marco-m** [comment]
As a happy `task` user, I am concerned that this proposal would make `task` into a kitchen sink.

**Comment by andreynering** [comment]
Hi @sawadashota,

I think a possible integration with direnv would be interesting. On the other side, people can already use it, they just need to install it manually.

The same for Mage, you can already call it manually inside Task, you just have the con of having to install yet another tool.

About others, I'd need a more detailed explanation on what these tools do that you can't make with Task. I don't have any experience with Ansible/Puppet/Ant/Sup. Sorry.

---

This kind of feature request raises a philosophical question, as @marco-m pointed out.

Task was initially created to be a generic build tool. It isn't supposed to do everything you may ever need. It is meant to have some basic generic features, and a great integration with external CLI tools you may found useful.

So, in short, if a given feature is simple and generic enough, and would be useful to more people and not only one or two users, I'll consider to implement it. Making everything Ansible does? Probably not.

FYI, there's a "preconditions" feature coming soon that will be useful to specific external dependencies for your scripts: https://github.com/go-task/task/pull/205.

**Comment by andreynering** [comment]
Hey @stephencheng,

I'm closing this issue for lack of feedback, since I haven't heard a word from you since then.

As I said, your features requests are too broad, and to me, as I never used the tools you mentioned (Ansible/Puppet/etc), it's also difficult to understand.

If you still think these feature would be useful, please open more detailed issues with examples, so I can properly understand you use case and proposed solution.

**Comment by stephencheng** [comment]
@andreynering @marco-m 

Following up my original request, I created this: https://github.com/upcmd/up

Thanks for inspiring!
